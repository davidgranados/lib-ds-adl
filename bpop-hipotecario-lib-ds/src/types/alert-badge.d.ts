import { ReactNode } from "react";
import { Testable } from "./global";

export interface AlertBadgeProps extends Testable {
  children?: ReactNode;
  type?: "info" | "warning" | "success" | "danger";
}
