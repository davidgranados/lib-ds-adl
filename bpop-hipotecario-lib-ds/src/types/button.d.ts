import { HTMLPropsFor } from "./component";

export type ButtonVariant = "primary" | "secondary";
export type ButtonTextSize = "sm" | "md" | "lg";
export type ButtonSize = "xs" | ButtonTextSize;

export interface DefaultButtonProps extends HTMLPropsFor<"button"> {
  label: string;
  variant?: ButtonVariant;
  disabled?: boolean;
  onClick?: () => void;
}

export interface ButtonProps extends DefaultButtonProps {
  size?: ButtonSize;
}

export interface ButtonTextProps extends DefaultButtonProps {
  size?: ButtonTextSize;
}
