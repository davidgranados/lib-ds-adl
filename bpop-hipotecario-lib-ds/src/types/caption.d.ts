import { ReactNode } from "react";
import { Testable } from "./global";

export declare type InputFieldCaptionStatus = "default" | "error" | "warning" | "success";

export declare interface InputFieldCaptionProps extends Testable {
  children: ReactNode;
  status?: InputFieldCaptionStatus;
}
