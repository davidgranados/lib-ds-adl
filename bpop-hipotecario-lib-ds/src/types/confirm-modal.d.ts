import { Testable } from "./global";

export declare interface ConfirmModalProps extends Testable {
  title: string;
  description?: string;
  show: boolean;
  mainButtonText?: string;
  secondaryButtonText?: string;
  onMainButtonClick?: () => void;
  onSecondaryButtonClick?: () => void;
  onToggle?: (state: boolean) => void;
}
