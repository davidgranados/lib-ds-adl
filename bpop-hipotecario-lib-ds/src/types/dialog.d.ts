import { Testable } from "./global";

export declare interface DialogProps extends Testable {
  show: boolean;
  width?: number;
  onToggle?: (state: boolean) => void;
}
