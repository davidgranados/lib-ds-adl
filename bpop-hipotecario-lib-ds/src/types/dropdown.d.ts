import { Testable } from "./global";

export declare type DropdownItems = { label: string; value: string | number }[];
export declare type DropdownOnChange = (v: string | number) => void;
export declare type DropdownValue = string | number;

export declare interface DropdownProps extends Testable {
  label: string;
  items: DropdownItems;
  disabled?: boolean;
  hasError?: boolean;
  caption?: string;
  placeholder?: string;
  value?: DropdownValue;
  onChange: DropdownOnChange;
}
