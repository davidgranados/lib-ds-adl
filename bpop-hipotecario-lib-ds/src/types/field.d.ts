import { ChangeEventHandler, FocusEventHandler, FormEventHandler } from "react";

import { InputBaseSize, InputBaseStatus } from "../../atoms/input-base";
import { TypographyStatus } from "../../atoms/typography";

export interface DefaultFieldProps {
  id: string;
  label: string;
  value?: string | number;
  status?: InputBaseStatus | TypographyStatus;
  caption?: string;
  placeholder?: string;
  size?: InputBaseSize;
  disabled?: boolean;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onInput?: FormEventHandler<HTMLInputElement>;
}
