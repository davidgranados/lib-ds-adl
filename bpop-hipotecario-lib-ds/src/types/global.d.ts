import React from "react";

export type HTMLPropsFor<Tag extends keyof JSX.IntrinsicElements> = JSX.IntrinsicElements[Tag];
export type CombinedRef = Array<React.MutableRefObject<null> | React.ForwardedRef<HTMLInputElement>>;
