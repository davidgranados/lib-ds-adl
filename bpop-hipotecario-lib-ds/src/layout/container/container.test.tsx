import React from "react";
import { mount, ReactWrapper } from "enzyme";

import { ContainerProps } from "../../types";

import Container from "./container";

describe("<Container />", () => {
  let wrapper: ReactWrapper<typeof Container>;
  const props: ContainerProps = {
    className: "class-from-implementation",
    title: "This is the title",
    titleClassName: "title-class-from-implementation",
    titleHTMLTag: "h6",
  };

  beforeEach(() => {
    jest.clearAllMocks();
    // given
    wrapper = mount<typeof Container>(<Container {...props}>Im in a Container</Container>);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
