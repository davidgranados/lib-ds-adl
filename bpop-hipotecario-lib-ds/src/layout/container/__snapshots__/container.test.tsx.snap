// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<div
  className="default class-from-implementation"
>
  <h6
    className="title title-class-from-implementation"
  >
    This is the title
  </h6>
  Im in a Container
</div>
`;
